src/main/java/org/osc/controller/nsc/api/NeutronSdnRedirectionApi.java

```diff
@@ -30,57 +36,78 @@
 import org.osc.sdk.controller.element.NetworkElement;
 import org.osc.sdk.controller.element.VirtualizationConnectorElement;
 import org.osc.sdk.controller.exception.NetworkPortNotFoundException;
+import org.osgi.service.transaction.control.TransactionControl;
 
 public class NeutronSdnRedirectionApi implements SdnRedirectionApi {
 
-    Logger log = Logger.getLogger(NeutronSdnRedirectionApi.class);
-
+	private static final Logger LOGGER = Logger.getLogger(NeutronSdnRedirectionApi.class);
+	
     private VirtualizationConnectorElement vc;
     private String region;
+    
+    private TransactionControl txControl;
+    private EntityManager em;
+    private NSCUtils utils;
 
     public NeutronSdnRedirectionApi() {
     }
 
-    public NeutronSdnRedirectionApi(VirtualizationConnectorElement vc, String region) {
+    public NeutronSdnRedirectionApi(VirtualizationConnectorElement vc, 
+    								String region, 
+    								TransactionControl txControl, 
+    								EntityManager em) {
         this.vc = vc;
         this.region = region;
+        this.txControl = txControl;
+        this.em = em;
+        this.utils = new NSCUtils(em, txControl);
     }
 
     @Override
-    public InspectionHookElement getInspectionHook(NetworkElement inspectedPort,
-            InspectionPortElement inspectionPort) throws NetworkPortNotFoundException, Exception {
-        try (NeutronSecurityControllerApi neutronApi = new NeutronSecurityControllerApi(new Endpoint(this.vc))){
-            return neutronApi.getInspectionHookByPorts(this.region, inspectedPort.getElementId(), inspectionPort);
+    public InspectionHookElement getInspectionHook(NetworkElement inspectedPort, InspectionPortElement inspectionPort)
+            throws Exception {
+//        try (NeutronSecurityControllerApi neutronApi = new NeutronSecurityControllerApi(new Endpoint(this.vc))) {
+//            return neutronApi.getInspectionHookByPorts(this.region, inspectedPort.getElementId(), inspectionPort);
+//        }
+    	CriteriaBuilder cb = this.em.getCriteriaBuilder();
+    	String inspectedPortId = (inspectedPort != null? inspectedPort.getElementId() : null);
+        CriteriaQuery<InspectionHookNSCEntity> q = cb.createQuery(InspectionHookNSCEntity.class);
```

Comment:
indentation, please configure the eclipse settings to automatically indent the file: spaces, extra lines, etc